File: Java/LeetCode/Min Max Game.java
// https://leetcode.com/problems/min-max-game

class Solution {

    public int minMaxGame(int[] nums) {
        int n = nums.length;
        if (n == 1) {
            return nums[0];
        }

        int[] newNums = new int[n/2];
        for (int i = 0; i < n/2; i += 2) {
            newNums[i] = Math.min(nums[2*i], nums[2*i+1]);
        }
        for (int i = 1; i < n/2; i += 2) {
            newNums[i] = Math.max(nums[2*i], nums[2*i+1]);
        }

        nums = newNums;
        
        return minMaxGame(nums);
    }

}

File: Kotlin/LeetCode/Min Max Game.kt
// https://leetcode.com/problems/min-max-game

class Solution {

    fun minMaxGame(nums: IntArray): Int {
        val n = nums.size
        if (n == 1) {
            return nums[0]
        }

        val newNums = IntArray(n/2)
        for (i in 0 until n/2 step 2) {
            newNums[i] = Math.min(nums[2*i], nums[2*i+1]);
        }
        for (i in 1 until n/2 step 2) {
            newNums[i] = Math.max(nums[2*i], nums[2*i+1]);
        }

        return minMaxGame(newNums)
    }

}
